app/helpers/validated_field_helper.rb

```diff
@@ -0,0 +1,43 @@
+module ValidatedFieldHelper
+  def form_errors(errors)
+    tag.ul(class: %w[help-block error-block]) do
+      safe_join(errors.map { |e| tag.li(e) }) if errors.present?
+    end
+  end
+
+  def form_label(id, label, classes: nil)
+    return nil unless id.present? && label.present?
+
+    tag.div(class: "form-label") do
+      tag.label(for: "edition_#{id}", class: classes) { label }
+    end
+  end
+
+  def form_help(help)
+    return nil if help.blank?
+
+    tag.div(class: "help-block") { help }
+  end
+
+  def form_input(classes: nil, &block)
+    klasses = %w[form-wrapper]
+    klasses << classes.flatten if classes
+
+    tag.div(class: klasses, &block)
+  end
+
+  def form_group(errors, field_name, label: nil, help: nil, classes: nil, label_classes: nil, &block)
+    klasses = %w[form-group]
+    klasses << "has-error" if errors[field_name] && errors[field_name].any?
+    klasses << classes.flatten if classes
+
+    tag.div(class: klasses) do
+      label = form_label(field_name, label, classes: label_classes)
+      field = form_input(&block)
+      help = form_help(help)
+      errors = form_errors(errors[field_name])
+
+      safe_join([label, field, help, errors])
+    end
+  end
```

Comment:
This stuff looks great 👍 

I've had a play around and have some suggestions to make for it.

Suggested approach: 

```ruby
module FormHelper
  def form_errors(errors)
    tag.ul(class: %w[help-block error-block]) do
      safe_join(errors.map { |e| tag.li(e) }) if errors.present?
    end
  end

  def form_group(form, field_name, label: nil, help: nil, attributes: {}, &block)
    errors = form.object.errors[field_name]
    attributes[:class] = Array(attributes[:class]) << "form-group"
    attributes[:class] << "has-error" if errors.any?

    tag.div(attributes) do
      wrapped_label = tag.div(class: "form-label") { form_label_element(form, field_name, label) }
      wrapped_field = tag.div(class: "form-wrapper", &block)
      errors = form_errors(errors)
      help = tag.div(class: "help-block") { help } if help

      safe_join([wrapped_label, wrapped_field, errors, help])
    end
  end

private

  def form_label_element(form, field_name, label)
    return label if label&.match?(/\A<label/) # already a label element

    form.label(field_name, label)
  end
end
```

Differences and why:

- form_group receives a form object rather than an errors block, we can use the form to create the label and access the errors
- label argument can be a HTML label, it means we can just create the label in the code if we need to rather than needing to pass attributes separately
- attributes argument, this can be used to pass through any HTML attribute tweaks we want to make to form group, saving a need for specific id argument too
- no label_classes argument, not necessary if label can be given classes directly
- no form_label helper, this helper seemed a bit confusing since it didn't return a label but a label wrapped in a div, also seemed coupled to editions
- no form_input or form_help helpers, these seemed pretty trivial and input seemed a little misleadingly named
- this will always generate a label element as it'll fall back to field name without them; this seems closer to how Rails behaves and I think fields that are complex enough to do label differently should be handled separately - this stuff should only be for the more conventional fields
- errors before help, this seemed to look nicer to me when testing, it seemed confusing to have field, help text then error

Applied to some examples this comes out like:

```erb
# simple
<%= form_group(f, :licence_identifier) do %>
  <%= f.text_field :licence_identifier, disabled: @resource.locked_for_edits?, class: "input-md-3 form-control" %>
<% end %>

# with help
<%= form_group(f, :continuation_link, label: "Link to competent authority", help: "Link as deep as possible.") do %>
  <%= f.text_field :continuation_link, disabled: @resource.locked_for_edits?, class: "input-md-7 form-control" %>
<% end %>

# complex
<%= form_group(
  f,
  :alternative_url,
  label: f.label(:alternative_url, "Enter the URL of the devolved administration website page", class: "inset-text-field-label"),
  attributes: { class: "inset-text-field" }
) do %>
  <%= f.text_field :alternative_url, disabled: @resource.locked_for_edits?, class: "input-md-7 form-control" %>
<% end %>
```

Reply:
Have update as per your suggestion (with a couple of minor tweaks).  Thanks